lib/tools.go

```diff
@@ -55,17 +59,33 @@ func init() {
 // Log
 func Log(f string, a ...interface{}) {
 	if ergoTrace {
-		log.Printf(f, a...)
+		richPrintf(f, a...)
 	}
 }
 
 // Warning
 func Warning(f string, a ...interface{}) {
 	if ergoWarning {
-		log.Printf("WARNING! "+f, a...)
+		richPrintf("WARNING! "+f, a...)
 	}
 }
 
+// Print more log information
+func richPrintf(formating string, args ...interface{}) {
```

Comment:
I would rather have Debug method with extended details. Log/Warning methods are not for that.

Reply:
For sure, this kind of update must be in a separate PR. This update should also introduce `ergoDebug` option for that purpose.

Reply:
You are right, I tried to introduce the `ergoDebug` option.

Reply:
I would really like to have goroutine id, line number, file name, function name, etc. in the log so that it will be very easy to debug.